hera_cal/abscal.py

```diff
@@ -632,60 +793,216 @@ def TT_Phi_gain_arr(self):
+    a.add_argument("--abs_amp_cal", default=False, action='store_true', help='perform absolute amplitude calibration')
+    a.add_argument("--TT_phs_cal", default=False, action='store_true', help='perform Tip-Tilt phase slope calibration')
+    a.add_argument("--gen_amp_cal", default=False, action='store_true', help='perform general antenna amplitude bandpass calibration')
+    a.add_argument("--gen_phs_cal", default=False, action='store_true', help='perform general antenna phase bandpass calibration')
+    return a
+
+
+def abscal_run(data_files, model_files, pol_select=None, verbose=True, overwrite=False,
+               write_calfits=True, calfits_fname=None, return_gains=False, outdir=None,
+               match_red_bls=False, tol=1.0, reweight=False, interp_model=True, alt_gains=False,
+               delay_slope_cal=True, abs_amp_cal=True, TT_phs_cal=True,
+               delay_cal=False, avg_phs_cal=False, gen_amp_cal=False, gen_phs_cal=False):
+    """
+    run AbsCal on a set of time-contiguous data miriad files, using 
+    time-contiguos model miriad files that cover the data_files across LST
 
+    Parameters:
+    -----------
+    data_files : type=list or string, path(s) to data miriad file(s)
+                a list of paths to miriad file(s) containing complex
+                visibility data, or a path itself
 
-        Parameters:
-        -----------
-        data : 
+    model_files : type=list or sring, path(s) to model miriad files(s)
+                a list of paths to miriad file(s) containing complex
+                visibility data, or a path itself
 
+    verbose : type=boolean, if True print output to stdout
 
-        flags : 
-    
-        """
-        # configure parameters
+    pol_select : type=list, list of polarization strings to use. Ex ['xx']
+
+    overwrite : type=boolean, if True, overwite output files
+
+    write_calfits : type=boolean, if True, write out gains as calfits file
 
+    calfits_fname : type=str, filename of output calfits filename
 
+    outdir : type=str, path to output directory
 
-        Ntimes
-        Nfreqs
+    return_gains : type=boolean, if True, return AbsCal gain dictionary
 
-        # interpolate flagged data
+    match_red_bls : type=boolean, match unique data baselines to model baselines based on redundancy
 
+    reweight : type=boolean, reweight unique baseline data based on redundancy
 
-        # sphere training data x-values
+    interp_model : type=boolean, interpolate model freq and time onto data freq and time
 
+    alt_gains : type=boolean, if True, use full antenna list in data file to make gains,
+                rather than just antennas present in the raw data. It is not possible
+                to run delay_cal, avg_phs_cal, gen_phs_cal and gen_amp_cal when alt_gains is True.
 
-        # ravel training data
+    delay_slope_cal : type=boolean, if True, perform delay slope calibration
```

Comment:
These booleans (which kinds of cal to run) should be in a separate section of the docstring from the other parameters. It should be clear that these calibrations are *both* run and applied. Also, they should appear exactly in the same order as they are run, to make it easier to understand what exactly is happening.